example/src/main/java/com/stripe/example/activity/AndroidPayActivity.java

```diff
@@ -0,0 +1,311 @@
+        }
+
+        String id = paymentSource.getId();
+
+        if (cardDetails == null) {
+            cardDetails = "UNKN";
+        }
+        mListViewController.addToList(cardDetails, id);
+    }
+
+    private void confirmPayment() {
+        mChangeDetailsContainer.setVisibility(View.GONE);
+        mConfirmDetailsContainer.setVisibility(View.VISIBLE);
+    }
+
+    private void proceedWithWallet(@NonNull MaskedWallet maskedWallet) {
+        mChangeDetailsContainer.setVisibility(View.GONE);
+
+        FullWalletRequest walletRequest =
+                AndroidPayConfiguration.generateFullWalletRequest(
+                        maskedWallet.getGoogleTransactionId(),
+                        mCart);
+        mProgressDialogController.startProgress();
+        loadFullWallet(walletRequest);
+    }
+
+    private void updateCartTotals(@NonNull CartManager cartManager) {
+        Long itemTotal = cartManager.calculateRegularItemTotal();
+        Long shippingTotal = cartManager.calculateShippingItemTotal();
+        Long tax = cartManager.calculateTax();
+        mItemsPriceDisplay.setText(
+                String.format(LOC,
+                        "Item Total: %s",
+                        getPriceString(itemTotal, DOLLARS)));
+        mShippingDisplay.setText(
+                String.format(LOC,
+                        "Shipping: %s",
+                        getPriceString(shippingTotal, DOLLARS)));
+        mTaxesDisplay.setText(String.format(LOC, "Tax: %s", getPriceString(tax, DOLLARS)));
+        mTotalPaymentDisplay.setText(String.format(
+                LOC,
+                "Total: %s",
+                getPriceString(addIfNotNull(itemTotal, shippingTotal, tax), DOLLARS)));
+    }
+
+    private void updatePaymentInformation(@NonNull MaskedWallet maskedWallet) {
+        if (maskedWallet.getPaymentDescriptions() != null
+                && maskedWallet.getPaymentDescriptions().length > 0) {
+            String cardText = String.format(LOC, "Card: %s", maskedWallet.getPaymentDescriptions()[0]);
+            mSelectedCardDisplay.setText(cardText);
+        }
+
+        CartManager copyCart = new CartManager(mCart);
+        long shippingUpdate = calculateShipping(maskedWallet.getBuyerShippingAddress());
+        long taxUpdate = calculateTaxes(maskedWallet.getBuyerBillingAddress());
+        copyCart.addShippingLineItem("Shipping", shippingUpdate);
+        copyCart.setTaxLineItem("Tax", taxUpdate);
+
+        if (maskedWallet.getBuyerShippingAddress() != null) {
+            mSelectedShippingAddressDisplay.setText(maskedWallet.getBuyerShippingAddress().getAddress1());
+        }
+
+        try {
+            mCart = copyCart.buildCart();
+            updateCartTotals(copyCart);
+        } catch (CartContentException unexpected) {
+            // ignore for now
+        }
+    }
+
+    private long calculateShipping(UserAddress userAddress) {
+        if (userAddress == null) {
```

Comment:
I can switch between values, though I expect a lot of these will be in-US. I can add some comments on the matter. The main purpose I wanted to show was why you might want to do the copy-all-things-but-shipping-and-taxes operation that occurs when you instantiate a new CartManager with an existing Cart instance, so that you can use the shipping address that you get back to do something.